gph/bindings/ripser_bindings.cpp

```diff
@@ -15,85 +15,84 @@
+          if (d <= threshold) ++num_edges;
+        }
 
-            ripserResults res;
-            ripser<compressed_lower_distance_matrix> r(std::move(dist), dim_max,
-                                                       threshold, ratio,
-                                                       modulus, num_threads);
-            r.compute_barcodes();
-            r.copy_results(res);
-            res.num_edges = num_edges;
-            return res;
-        },
-        "D"_a, "N"_a, "modulus"_a, "dim_max"_a, "threshold"_a, "num_threads"_a,
-        "ripser distance matrix");
+        ripserResults res;
+        ripser<compressed_lower_distance_matrix> r(std::move(dist), dim_max,
+                                                   threshold, ratio, modulus,
+                                                   num_threads, do_cocycles);
+        r.compute_barcodes();
+        r.copy_results(res);
+        res.num_edges = num_edges;
+        return res;
+      },
+      "D"_a, "N"_a, "modulus"_a, "dim_max"_a, "threshold"_a, "num_threads"_a,
+      "do_cocycles"_a, "ripser distance matrix");
 
-    m.def(
-        "rips_dm_sparse",
-        [](py::array_t<index_t>& I, py::array_t<index_t>& J,
-           py::array_t<value_t>& V, int NEdges, int N, int modulus, int dim_max,
-           float threshold, int num_threads) {
-            auto I_ = static_cast<index_t*>(I.request().ptr);
-            auto J_ = static_cast<index_t*>(J.request().ptr);
-            auto V_ = static_cast<value_t*>(V.request().ptr);
+  m.def(
+      "rips_dm_sparse",
+      [](py::array_t<index_t>& I, py::array_t<index_t>& J,
+         py::array_t<value_t>& V, int NEdges, int N, int modulus, int dim_max,
+         float threshold, int num_threads, bool do_cocycles) {
+        auto I_ = static_cast<index_t*>(I.request().ptr);
+        auto J_ = static_cast<index_t*>(J.request().ptr);
+        auto V_ = static_cast<value_t*>(V.request().ptr);
 
-            // TODO: This seems like a dummy parameter at the moment
-            const float ratio = 1.0;
-            // Setup distance matrix and figure out threshold
-            ripser<sparse_distance_matrix> r(
-                sparse_distance_matrix(I_, J_, V_, NEdges, N, threshold),
-                dim_max, threshold, ratio, modulus, num_threads);
-            r.compute_barcodes();
-            // Report the number of edges that were added
-            int num_edges = 0;
-            for (int idx = 0; idx < NEdges; idx++) {
-                if (I_[idx] < J_[idx] && V_[idx] <= threshold) {
-                    num_edges++;
-                }
-            }
-            ripserResults res;
-            r.copy_results(res);
-            res.num_edges = num_edges;
-            return res;
-        },
-        "I"_a, "J"_a, "V"_a, "NEdges"_a, "N"_a, "modulus"_a, "dim_max"_a,
-        "threshold"_a, "num_threads"_a, "ripser sparse distance matrix");
+        // TODO: This seems like a dummy parameter at the moment
```

Comment:
As above for `ratio` and `num_edges`

Reply:
I added a note directly into the project 0.2, we could maybe resolve the conversations ?

Reply:
Great!  So we address these in another PR, and I resolve here.